spire-agent validate panicing on latest master

<!--
If you are looking for help, please join us on [Slack or the mailing
lists.](https://github.com/spiffe/spiffe/blob/master/README.md)

If you are submitting a feature request or bug report, please be
as detailed as possible.
-->

* **Version**: 3eac3b098054417d667e279229ab901aca1cd5ee
* **Platform**: docker linux
* **Subsystem**: spire-agent validate config

The line in question from the stack below is this:
```
	ac.DefaultSVIDName = c.Agent.SDS.DefaultSVIDName
```
so I'm guessing `c.Agent.SDS` is nil, but haven't confirmed yet

```
spire-agent validate -config agent-test.conf
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x106c36f]

goroutine 1 [running]:
github.com/spiffe/spire/cmd/spire-agent/cli/run.NewAgentConfig(0xc0000f5200, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00044f320)
	/src/vendor/github.com/spiffe/spire/cmd/spire-agent/cli/run/run.go:282 +0x40f
github.com/spiffe/spire/cmd/spire-agent/cli/validate.(*validateCommand).run(0xc000451140, 0xc00003c0e0, 0x2)
	/src/vendor/github.com/spiffe/spire/cmd/spire-agent/cli/validate/validate.go:67 +0x72
github.com/spiffe/spire/cmd/spire-agent/cli/validate.(*validateCommand).Run(0xc000451140, 0xc00003c0e0, 0x2, 0x2, 0x1411520)
	/src/vendor/github.com/spiffe/spire/cmd/spire-agent/cli/validate/validate.go:44 +0x76
github.com/mitchellh/cli.(*CLI).Run(0xc00064f400, 0xc0001374d0, 0x139289c, 0x8)
	/src/vendor/github.com/mitchellh/cli/cli.go:255 +0x1da
github.com/spiffe/spire/cmd/spire-agent/cli.(*CLI).Run(0xc000450d00, 0xc00003c0d0, 0x3, 0x3, 0x5ce00)
	/src/vendor/github.com/spiffe/spire/cmd/spire-agent/cli/cli.go:49 +0x3cf
main.main()
	/src/spire/cmd/spire-agent/agent.go:13 +0xb3
```

--- a/cmd/spire-agent/cli/run/run.go
+++ b/cmd/spire-agent/cli/run/run.go
@@ -48,19 +48,19 @@ type Config struct {
 }
 
 type agentConfig struct {
-	DataDir           string     `hcl:"data_dir"`
-	EnableSDS         bool       `hcl:"enable_sds"`
-	InsecureBootstrap bool       `hcl:"insecure_bootstrap"`
-	JoinToken         string     `hcl:"join_token"`
-	LogFile           string     `hcl:"log_file"`
-	LogFormat         string     `hcl:"log_format"`
-	LogLevel          string     `hcl:"log_level"`
-	SDS               *sdsConfig `hcl:"sds"`
-	ServerAddress     string     `hcl:"server_address"`
-	ServerPort        int        `hcl:"server_port"`
-	SocketPath        string     `hcl:"socket_path"`
-	TrustBundlePath   string     `hcl:"trust_bundle_path"`
-	TrustDomain       string     `hcl:"trust_domain"`
+	DataDir           string    `hcl:"data_dir"`
+	EnableSDS         bool      `hcl:"enable_sds"`
+	InsecureBootstrap bool      `hcl:"insecure_bootstrap"`
+	JoinToken         string    `hcl:"join_token"`
+	LogFile           string    `hcl:"log_file"`
+	LogFormat         string    `hcl:"log_format"`
+	LogLevel          string    `hcl:"log_level"`
+	SDS               sdsConfig `hcl:"sds"`
+	ServerAddress     string    `hcl:"server_address"`
+	ServerPort        int       `hcl:"server_port"`
+	SocketPath        string    `hcl:"socket_path"`
+	TrustBundlePath   string    `hcl:"trust_bundle_path"`
+	TrustDomain       string    `hcl:"trust_domain"`
 
 	ConfigPath string
 	ExpandEnv  bool
@@ -394,7 +394,7 @@ func defaultConfig() *Config {
 			LogLevel:   defaultLogLevel,
 			LogFormat:  log.DefaultFormat,
 			SocketPath: defaultSocketPath,
-			SDS: &sdsConfig{
+			SDS: sdsConfig{
 				DefaultBundleName: defaultDefaultBundleName,
 				DefaultSVIDName:   defaultDefaultSVIDName,
 			},

--- a/cmd/spire-agent/cli/run/run_test.go
+++ b/cmd/spire-agent/cli/run/run_test.go
@@ -154,7 +154,7 @@ func TestMergeInput(t *testing.T) {
 		{
 			msg: "default_svid_name should be configurable by file",
 			fileInput: func(c *Config) {
-				c.Agent.SDS = &sdsConfig{
+				c.Agent.SDS = sdsConfig{
 					DefaultSVIDName: "foo",
 				}
 			},
@@ -174,7 +174,7 @@ func TestMergeInput(t *testing.T) {
 		{
 			msg: "default_bundle_name should be configurable by file",
 			fileInput: func(c *Config) {
-				c.Agent.SDS = &sdsConfig{
+				c.Agent.SDS = sdsConfig{
 					DefaultBundleName: "foo",
 				}
 			},

